pkg/operator2/workload/sync_openshift_oauth_apiserver.go

```diff
@@ -0,0 +1,261 @@
+	if err != nil {
+		errs = append(errs, err)
+		return nil, errs
+	}
+
+	// TODO: block until config is obvserved when required
+	/*if operatorCfg, err := getStructuredConfig(authOperator.Spec.OperatorSpec); err != nil {
+		errs = append(errs, err)
+		return nil, errs
+	} else {
+		if len(operatorCfg.APIServerArguments) == 0 {
+			klog.Info("Waiting for observed configuration to be available")
+			errs = append(errs, errors.New("waiting for observed configuration to be available (spec.ObservedConfig.Raw)"))
+			return nil, errs
+		}
+	}*/
+
+	// manage assets
+	directResourceResults := resourceapply.ApplyDirectly(c.kubeClient, c.eventRecorder, assets.Asset,
+		"oauth-apiserver/ns.yaml",
+		"oauth-apiserver/apiserver-clusterrolebinding.yaml",
+		"oauth-apiserver/svc.yaml",
+		"oauth-apiserver/sa.yaml",
+		"oauth-apiserver/cm.yaml",
+	)
+	for _, currResult := range directResourceResults {
+		if currResult.Error != nil {
+			errs = append(errs, fmt.Errorf("%q (%T): %v", currResult.File, currResult.Type, currResult.Error))
+		}
+	}
+
+	actualDaemonSet, err := c.syncDaemonSet(authOperator, authOperator.Status.Generations)
+	if err != nil {
+		errs = append(errs, fmt.Errorf("%q: %v", "daemonsets", err))
+	}
+	return actualDaemonSet, errs
+}
+
+func (c *OAuthAPIServerWorkload) syncDaemonSet(authOperator *operatorv1.Authentication, generationStatus []operatorv1.GenerationStatus) (*appsv1.DaemonSet, error) {
+	tmpl, err := assets.Asset("oauth-apiserver/ds.yaml")
+	if err != nil {
+		return nil, err
+	}
+
+	operatorCfg, err := getStructuredConfig(authOperator.Spec.OperatorSpec)
+	if err != nil {
+		return nil, err
+	}
+
+	// log level verbosity is taken from the spec always
+	operatorCfg.APIServerArguments["v"] = []string{loglevelToKlog(authOperator.Spec.LogLevel)}
+	operandFlags := configflags.ToFlagSlice(operatorCfg.APIServerArguments)
+
+	r := strings.NewReplacer(
+		"${IMAGE}", c.targetImagePullSpec,
+		// TODO: add LatestAvailableRevision support
+		//"${REVISION}", strconv.Itoa(int(authOperator.Status.LatestAvailableRevision)),
+		"${REVISION}", "1",
+		"${FLAGS}", strings.Join(padFlags(operandFlags, strings.Repeat(" ", 14)), " \\\n"),
+	)
+
+	tmpl = []byte(r.Replace(string(tmpl)))
+	re := regexp.MustCompile("\\$\\{[^}]*}")
+	if match := re.Find(tmpl); len(match) > 0 {
+		return nil, fmt.Errorf("invalid template reference %q", string(match))
+	}
+
+	required := resourceread.ReadDaemonSetV1OrDie(tmpl)
+
+	// we set this so that when the requested image pull spec changes, we always have a diff.  Remember that we don't directly
+	// diff any fields on the daemonset because they can be rewritten by admission and we don't want to constantly be fighting
+	// against admission or defaults.  That was a problem with original versions of apply.
+	if required.Annotations == nil {
+		required.Annotations = map[string]string{}
+	}
+	required.Annotations["openshiftapiservers.operator.openshift.io/pull-spec"] = c.targetImagePullSpec
```

Comment:
what does this mean for multi-operand operators?

Reply:
the operator will create a separate instance for each controller and will pass distinct images, for example - https://github.com/openshift/cluster-authentication-operator/pull/243/files#diff-c9f95324de2319f30fff1ad2fd4e7b00R231

Reply:
oauthapiservers.operator.openshift.io ?